src/main/resources/assets/admin/common/js/ui/dialog/ModalDialog.ts

```diff
@@ -806,4 +812,108 @@ module api.ui.dialog {
         });
     }
 
+    class DialogManagerInner {
+
+        private openDialogs: ModalDialog[];
+
+        private maskedBy: Map<string, ModalDialog[]>;
+
+        private dialogOpenListeners: { (dialog: ModalDialog): void; } [];
+
+        private static INSTANCE: DialogManagerInner;
+
+        private constructor() {
+            this.openDialogs = [];
+            this.dialogOpenListeners = [];
+            this.maskedBy = new Map();
+        }
+
+        public static get(): DialogManagerInner {
+            if (!DialogManagerInner.INSTANCE) {
+                DialogManagerInner.INSTANCE = new DialogManagerInner();
+            }
+
+            return DialogManagerInner.INSTANCE;
+        }
+
+        opened(dialog: ModalDialog) {
```

Comment:
`opened` and `closed` are not good names for these methods. `handleOpenDialog` and `handleClosedDialog` should be fine.